src/main/java/fr/openmc/core/features/contest/managers/ContestPlayerManager.java

```diff
@@ -0,0 +1,287 @@
+        return contestManager.data.get("camp" + campInteger);
+    }
+
+    public Integer getOfflinePlayerCamp(OfflinePlayer player) {
+        String sql = "SELECT * FROM contest_camps WHERE minecraft_uuid = ?";
+        try (PreparedStatement states = DatabaseManager.getConnection().prepareStatement(sql)) {
+            states.setString(1, player.getUniqueId().toString());
+            ResultSet result = states.executeQuery();
+            if (result.next()) {
+                return result.getInt("camps");
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+        return 0;
+    }
+
+    public String getOfflinePlayerCampName(OfflinePlayer player) {
+        Integer campInteger = getOfflinePlayerCamp(player);
+        return contestManager.data.get("camp" + campInteger);
+    }
+
+    public ChatColor getOfflinePlayerCampChatColor(OfflinePlayer player) {
+        Integer campInteger = getOfflinePlayerCamp(player);
+        String color = contestManager.data.get("color" + campInteger);
+        return ChatColor.valueOf(color);
+    }
+
+    public void setPointsPlayer(Player player, int points) {
+        ContestManager manager = ContestManager.getInstance();
+        ContestPlayer data = manager.dataPlayer.get(player.getUniqueId().toString());
+
+        manager.dataPlayer.put(player.getUniqueId().toString(), new ContestPlayer(data.getName(), points, data.getCamp(), data.getColor()));
+    }
+
+    public Integer getRankPlayerInContest(Integer pointsDep) {
+        String sql = "SELECT COUNT(*) AS rank FROM contest_camps WHERE point_dep > ?";
+        try (PreparedStatement states = DatabaseManager.getConnection().prepareStatement(sql)) {
+            states.setInt(1, pointsDep);
+            ResultSet result = states.executeQuery();
+            if (result.next()) {
+                return result.getInt("rank") + 1;
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+        return -1;
+    }
+
+    public String getRankContest(Player player) {
+        int points = contestManager.dataPlayer.get(player.getUniqueId().toString()).getPoints();
+
+        if(points >= 10000) {
+            return "Dictateur en ";
+        } else if (points >= 2500) {
+            return "Colonel en ";
+        } else if (points >= 2000) {
+            return "Addict en ";
+        } else if (points >= 1500) {
+            return "Dieu en ";
+        } else if (points >= 1000) {
+            return "Légende en ";
+        } else if (points >= 750) {
+            return "Sénior en ";
+        } else if (points >= 500) {
+            return "Pro en ";
+        } else if (points >= 250) {
+            return "Semi-Pro en ";
+        } else if (points >= 100) {
+            return "Amateur en ";
+        } else if (points >= 0) {
+            return "Noob en ";
+        }
+
+        return "";
```

Comment:
Ceci est plus optimisé, mais pas plus clair
```suggestion
        int[] seuils = {10000, 2500, 2000, 1500, 1000, 750, 500, 250, 100, 0};
        String[] categories = {
            "Dictateur en ",
            "Colonel en ",
            "Addict en ",
            "Dieu en ",
            "Légende en ",
            "Sénior en ",
            "Pro en ",
            "Semi-Pro en ",
            "Amateur en ",
            "Noob en "
        };
        
        for (int i = 0; i < seuils.length; i++) {
            if (points >= seuils[i]) {
                return categories[i];
            }
        }

        return "";
```

Reply:
Je comptais faire une refonte de ceci, je me disais que ça fait moche et un peu bete de faire que des if